spec/controllers/hyrax/dashboard/collections_controller_spec.rb

```diff
@@ -163,31 +163,35 @@
             creator: [""]
           }
         }
-        expect(assigns[:change_set].resource.title).to eq ["My Next Collection "]
-        expect(assigns[:change_set].resource.creator).to eq []
+        expect(assigns[:resource].title).to eq ["My Next Collection "]
+        expect(assigns[:resource].creator).to eq []
       end
     end
 
-    context "when update fails" do
-      let(:collection) { Collection.new(id: '12345') }
-      let(:repository) { instance_double(Blacklight::Solr::Repository, search: result) }
-      let(:result) { double(documents: []) }
+    context 'when update fails' do
+      let(:collection) { Collection.new }
+      let(:change_set) do
+        instance_double(Hyrax::CollectionChangeSet,
+                        validate: false,
+                        resource: collection)
+      end
 
       before do
         allow(controller).to receive(:authorize!)
-        allow(Collection).to receive(:find).and_return(collection)
-        allow(collection).to receive(:update).and_return(false)
-        allow(controller).to receive(:repository).and_return(repository)
+        allow(controller).to receive(:query_collection_members)
+        allow(controller).to receive(:find_resource).and_return(collection)
+        allow(Hyrax::CollectionChangeSet).to receive(:new).and_return(change_set)
+        allow(change_set).to receive(:prepopulate!).and_return(change_set)
       end
 
-      it "renders the form again" do
+      it 'renders the form again' do
         put :update, params: {
           id: collection,
           collection: collection_attrs
         }
         expect(response).to be_successful
         expect(response).to render_template(:edit)
-        expect(assigns[:member_docs]).to be_kind_of Array
+        expect(controller).to have_received(:query_collection_members)
```

Comment:
I kind of like the test of `assigns[:member_docs]` because it describes the post condition of the controller action.  Despite the ugly mocking to support this I feel like it might still be better than only checking that a private method was called.

Reply:
Thanks for the feedback. I'll make that change.